File: inc.graphql-resources/backend/nodejs/models/core/notification.js
module.exports = (sequelize, DataTypes) => {
    const Notification = sequelize.define('notification', {
        notification_id: {
            type: DataTypes.STRING(40),
            primaryKey: true,
            allowNull: false
        },
        notification_type: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        admin_group: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        admin_id: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        icon: {
            type: DataTypes.STRING(40),
            allowNull: true
        },
        subject: {
            type: DataTypes.STRING(255),
            allowNull: true
        },
        content: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        link: {
            type: DataTypes.TEXT,
            allowNull: true
        },
        is_read: {
            type: DataTypes.BOOLEAN,
            allowNull: true
        },
        time_create: {
            type: DataTypes.DATE,
            allowNull: true
        },
        ip_create: {
            type: DataTypes.STRING(50),
            allowNull: true
        },
        time_read: {
            type: DataTypes.DATE,
            allowNull: true
        },
        ip_read: {
            type: DataTypes.STRING(50),
            allowNull: true
        }
    }, {
        tableName: 'notification',
        timestamps: false
    });

    Notification.associate = (models) => {
        Notification.belongsTo(models.admin, { foreignKey: 'admin_id' });
        Notification.belongsTo(models.admin_level, { foreignKey: 'admin_group' });
    };

    return Notification;
};

File: inc.graphql-resources/backend/python/models/core/notification.py
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, ForeignKey
from database import Base
from datetime import datetime

class Notification(Base):
    __tablename__ = 'notification'

    notification_id = Column(String(40), primary_key=True)
    admin_id = Column(Integer, ForeignKey('admin.admin_id'), nullable=True)
    admin_group = Column(String(40), ForeignKey('admin_level.admin_level_id'), nullable=True)
    subject = Column(String(255), nullable=False)
    content = Column(Text)
    link = Column(String(255))
    time_create = Column(DateTime, default=datetime.utcnow)
    is_read = Column(Boolean, default=False)
    time_read = Column(DateTime, nullable=True)
    ip_read = Column(String(50), nullable=True)